contracts/InsuranceFund.sol

```diff
@@ -60,6 +88,46 @@ contract InsuranceFund is IInsuranceFund, ReentrancyGuardUpgradeable, OwnerPausa
         emit Repaid(repaidAmount, tokenBalanceAfterRepaid);
     }
 
+    /// @inheritdoc IInsuranceFund
+    function distributeFee() external override nonReentrant whenNotPaused returns (uint256) {
+        address vault = _borrower;
+        address token = _token;
+        address surplusBeneficiary = _surplusBeneficiary;
+        int256 threshold = _threshold.toInt256();
+
+        // IF_SNS: surplusBeneficiary not yet set
+        require(surplusBeneficiary != address(0), "IF_SNS");
+
+        // IF_TEZ: threshold is equal to zero
+        require(threshold > 0, "IF_TEZ");
+
+        int256 insuranceFundFreeCollateral = IVault(vault).getFreeCollateralByToken(address(this), token).toInt256();
+
+        int256 insuranceFundCapacity = getInsuranceFundCapacity();
+
+        // We check IF's capacity against the threshold, which means if someone sends
+        // non-settlement collaterals to IF, it will increase IF's capacity and might help it meet the threshold.
+        // However, since only settlement collateral can be counted as surplus, in certain circumstances you may find
+        // that IF cannot distribute fee because it has zero surplus even though its capacity has met the threshold.
+        int256 overThreshold = PerpMath.max(insuranceFundCapacity.sub(threshold), 0);
+        uint256 surplus = PerpMath.min(overThreshold, insuranceFundFreeCollateral).toUint256();
+
+        if (surplus > 0) {
```

Comment:
Wonder why don't we require this and revert when the condition isn't met; no strong preference and would just like to know the reason!

Reply:
Can't remember the exact reason, maybe It's more convenient for the bot to decide whether to trigger fee distribution or more flexible if we want to build a wrapper contract in the future.